Frameworks/Foundation/NSOrderedSet.mm

```diff
@@ -66,84 +63,112 @@ + (instancetype)orderedSetWithObjects:(id)firstObj {
-    UNIMPLEMENTED();
-    return StubReturn();
++ (instancetype)orderedSetWithObjects:(id _Nonnull const[])objects count:(NSUInteger)count {
+    return [[[NSOrderedSet alloc] initWithObjects:objects count:count] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithOrderedSet:(NSOrderedSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [NSOrderedSet orderedSetWithOrderedSet:set range:NSMakeRange(0, [set count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithOrderedSet:(NSOrderedSet*)set range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSOrderedSet alloc] initWithOrderedSet:set range:range copyItems:flag] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithSet:(NSSet*)set {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSOrderedSet alloc] initWithSet:set] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 + (instancetype)orderedSetWithSet:(NSSet*)set copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [[[NSOrderedSet alloc] initWithSet:set copyItems:flag] autorelease];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:NO];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    return [self initWithArray:array range:NSMakeRange(0, [array count]) copyItems:flag];
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
 - (instancetype)initWithArray:(NSArray*)array range:(NSRange)range copyItems:(BOOL)flag {
-    UNIMPLEMENTED();
-    return StubReturn();
+    if (range.length > [array count]) {
+        [NSException raise:NSInvalidArgumentException format:@"The given range length is higher than the array length: %d.", [array count]];
+    }
+
+    if (self = [self init]) {
+        NSUInteger start = range.location;
+        NSUInteger end = range.length + range.location;
+
+        if (flag) {
+            for (NSUInteger i = start; i < end; i++) {
+                id obj = [[array objectAtIndex:i] copy];
+                [self _insertObject:obj];
+                [obj release];
+            }
+        } else {
+            for (NSUInteger i = start; i < end; i++) {
+                [self _insertObject:[array objectAtIndex:i]];
+            }
+        }
+    }
+    return self;
 }
 
+- (void)_insertObject:(id)object {
+    if ([self containsObject:object]) {
+        return;
+    }
+
+    setContainer.push_back(object);
+}
+
+- (NSUInteger)count {
+    return setContainer.size();
+}
+
+// - (NSArray*) array {
```

Comment:
indeed :)

---
In reply to: [61936592](https://github.com/Microsoft/WinObjC/pull/472#discussion_r61936592) [](ancestors = 61936592)